app/src/main/java/com/amaze/filemanager/filesystem/ftp/NetCopyConnectionInfo.kt

```diff
@@ -104,7 +109,18 @@ class NetCopyConnectionInfo(url: String) {
                      * Invalid string would have been trapped to other branches. Strings fell into
                      * this branch must be integer
                      */
-                        it[7].toInt()
+                        try {
+                            // Need to make sure port number is in range
+                            if (it[7].toInt() in 1..65535) {
+                                it[7].toInt()
+                            } else {
+                                LOGGER.warn("Port number is out of range: ${it[7]}")
+                                0
```

Comment:
Shouldn't this return the default port for ssh? I would think that returning 0 would just crash whatever uses this value?

Reply:
I prefer this to be the responsibility of downstream code - here is for just the regex extraction magic.

Reply:
Sorry for the delay.  The type looks like it doesn't have any error state. `port: Int` should probably be `port: Int?` and be set to null on these cases, so that users know that the port could be "broken/unset".